test/unit/parser-test.js

```diff
@@ -67,6 +67,20 @@ define([
             'returns a directory url ending with a forward slash "'+ scriptPath +'"');
     });
 
+    test('parser.getMediaId', function(assert) {
+        var path = parser.getMediaId(null);
+        assert.equal(path, null, 'returns null when no file name is provided');
+
+        var mediaId = parser.getMediaId('/test/with/short/mediaid.js');
+        assert.equal(mediaId, null, 'returns null when media is not 8 characters long');
+
+        mediaId = parser.getMediaId('/test/with/dash/mediaid1-dash.js');
+        assert.equal(mediaId, 'mediaid1', 'returns media id correctly with dash');
+
+        mediaId = parser.getMediaId('/test/with/mediaid2.js');
+        assert.equal(mediaId, 'mediaid2', 'returns media id correctly');
```

Comment:
Use real examples, with and without http(s):

//content.jwplatform.com/manifests/IMUjQRAB.m3u8
//content.jwplatform.com/videos/IMUjQRAB-injeKYZS.mp4
//content.jwplatform.com/videos/IMUjQRAB-rjiewRbX.ogg